### PR TITLE
Added Azure SSO in schema

### LINKED ISSUE #3849: Service Connection Schema - Prepare Azure JSON
Prepare Azure JSON Schema for the auth provider

### PR DESCRIPTION
**Pull request by ayush-shah**
### Describe your changes :

Worked on adding azure sso json schema
Note - Ingestion with sso is still broken
fix #3849 

#
### Type of change :

- [x] Bug fix
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

@harshach @pmbrull

### CODE REVIEW

**Comment by pmbrull** [comment]
@ayush-shah do we need to add the auth class as well right? https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/ometa/openmetadata_rest.py#L156 and use it https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/ingestion/ometa/ometa_api.py#L144

File: catalog-rest-service/src/main/resources/json/schema/metadataIngestion/workflow.json [hunk 1]
```diff
@@ -5,6 +5,32 @@
   "description": "OpenMetadata Ingestion Framework definition.",
   "type": "object",
   "definitions": {
+    "azureSSOConfig": {
+      "description": "Azure SSO client security configs.",
+      "type": "object",
+      "properties": {
+        "secretKey": {
```
Comment (akash-jain-10) [inline catalog-rest-service/src/main/resources/json/schema/metadataIngestion/workflow.json:12]: can we use the name as `clientSecret` here ? This is the standard term used by Azure!

File: catalog-rest-service/src/main/resources/json/schema/metadataIngestion/workflow.json [hunk 2]
```diff
@@ -5,6 +5,32 @@
...
+        },
+        "scopes": {
+          "description": "Azure Client ID.",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "additionalProperties": false,
+      "required": ["secretKey"]
```
Comment (akash-jain-10) [inline catalog-rest-service/src/main/resources/json/schema/metadataIngestion/workflow.json:32]: @ayush-shah - So, clientId, scopes, authority should also be required fields right ?

Comment (ayush-shah) [reply catalog-rest-service/src/main/resources/json/schema/metadataIngestion/workflow.json:32]: Yes - will add that.

File: catalog-rest-service/src/main/resources/json/schema/metadataIngestion/workflow.json (hunk 1 above)
Comment (ayush-shah) [reply catalog-rest-service/src/main/resources/json/schema/metadataIngestion/workflow.json:12]: Yes, will modify it likewise

**Review by pmbrull — APPROVED** [review]

**Comment by ayush-shah** [comment]
Closing this PR - will create a new one with all the required changes

**Pull request closed without merging**